### DISCUSSION TITLE
Prisma FindUnique on Child Relation

### DISCUSSION THREAD

**Discussion by fjur** [prisma/prisma#13886] [Q&A]
I have two models, model User has a one to many relation with model Post. I would like to get a unique record in model Post, but only where the parent relation is of a certain id. 

This is to simulate an api call such as `/user/:userId/post/:postId`. Get a post under a specific user

**How would you cleanly do this in Prisma?**

I thought this would work but it doesn't
```
const userId = 'userId'
const postId = 'postId'

prisma.post.findUnique({
  where:{
    id: postId,
    User: {
       id: userId
        }
     }
}
```

Schema
```
model User {
  id        String   @id @default(cuid())
  Post   Post[]
}

model Post {
  id            String     @id @default(cuid())
  User    Workspace @relation(fields: [userId], references: [id])
  userId   String
}
```

**Comment by l2D** [comment]
If your `id` of `Post` is unique, you could do this:

```js
prisma.post.findUnique({
    where: {
        id: postId,
    },
    include: {
        User: true
    },
}

// Next logic; (maybe check `id` from a query is equal with `userId` from params or not.)
```

But you could try this if you would like to select with both `userId` and `postId`:

```js
prisma.post.findFirst({
    where: {
        id: postId,
        User: {
            id: userId
        },
    }
}
```

**Reply by fjur** [reply]
What about for doing a similar query but with a delete? So delete a post that is under a specific user.

Want to be able to delete a post under a user, but seems like .delete you cannot chain the query, and there is deleteMany but it doesn't return the model that gets deleted, it returns the batch job, and risks deleting multiple if a field is empty by accident.

**Reply by l2D** [reply]
You could delete it with `postId` because it's unique.

```js
model Post {
  id        String      @id @default(cuid())
  User      Workspace   @relation(fields: [userId], references: [id])
  userId    String
}
```

**Reply by fjur** [reply]
Is there a way to enforce the restriction without having to do a query ahead of time to check?

You could delete via `postId` but I want to explicitly only delete if the post also has a relation to the user where `userId = some value`
```
 prisma.post.delete({
  where: {
    id: postId
  }
})
```

The above does not enforce that restriction and it seems like the only way to enforce that is to query `User` with an include with `Post` and do a loop to find if there is a matching post.

Any suggestions to do this efficiently? Ie a user can only delete a post they own.

**Reply by l2D** [reply]
Then, you could try the nested `delete` query. (ref: [Nested queries](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#delete-1))

E.g. Update an existing `User` record by deleting two `Post` records it's connected to

```js
const user = await prisma.user.update({
  where: { id: userId },
  data: {
    posts: {
      delete: [{ id: postId1 }, { id: postId2 }],
    },
  },
})
```

**Remarks:**
- delete is only available if the relation is **_optional_**.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
